cloud.foundry.cli/src/test/java/cloud/foundry/cli/system/GetAllCommandTest.java

```diff
@@ -0,0 +1,142 @@
+package cloud.foundry.cli.system;
+
+import cloud.foundry.cli.crosscutting.mapping.YamlMapper;
+import cloud.foundry.cli.crosscutting.mapping.beans.*;
+import cloud.foundry.cli.system.util.ArgumentsBuilder;
+import cloud.foundry.cli.system.util.RunResult;
+import cloud.foundry.cli.system.util.SpaceManager;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for {@link cloud.foundry.cli.services.GetController}, specifically the default get command which fetches all
+ * the information and outputs them in the same format we expect for the input.
+ */
+public class GetAllCommandTest extends SystemTestBase {
+
+    private void assertRootBeanIsValid(ConfigBean rootBean, SpaceManager spaceManager) {
+        assert rootBean.getApiVersion().startsWith("2.");
+
+        TargetBean target = rootBean.getTarget();
+        assert target.getEndpoint().equals(spaceManager.getCfApiEndpoint());
+        assert target.getOrg().equals(spaceManager.getCfOrganization());
+        assert target.getSpace().equals(spaceManager.getSpaceName());
+
+        SpecBean spec = rootBean.getSpec();
+        // TODO: check if we can put in an absolute value once we use "throw-away spaces" created per test run
+        assert spec.getSpaceDevelopers().size() > 0;
+    }
+
+    private ConfigBean loadConfigBeanFromStdoutAndAssertItsValidity(String stdoutContent, SpaceManager spaceManager) {
+        ConfigBean rootBean = YamlMapper.loadBeanFromString(stdoutContent, ConfigBean.class);
+        assertRootBeanIsValid(rootBean, spaceManager);
+        return rootBean;
+    }
+
+    @Test
+    public void testEmptySpace() {
+        try (SpaceManager spaceManager = createSpaceWithRandomName()) {
+            ArgumentsBuilder args = new ArgumentsBuilder()
+                    .addArgument("get");
+
+            RunResult runResult = runBaseControllerWithCredentialsFromEnvironment(args, spaceManager);
+
+            String outContent = runResult.getStreamContents().getStdoutContent();
+            assert outContent.length() > 0;
+
+            // parse YAML from stdout and check it for validity
+            ConfigBean rootBean = loadConfigBeanFromStdoutAndAssertItsValidity(outContent, spaceManager);
+
+            SpecBean spec = rootBean.getSpec();
+            assert spec.getApps() == null;
+            assert spec.getServices() == null;
+            assert spec.getSpaceDevelopers().size() > 0;
+
+            // TODO: check log contents
```

Comment:
Is it planned to address this TODO? It not we should remove it.

Reply:
We could perform a veeery basic test there, the exact logging is handled in regular unit tests.